RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,174 @@
+| refund_identity | The `pubkeyhash` of the refunding party                                                                                   |
+| expiry          | The absolute UNIX timestamp in seconds after which the HTLC can be refunded |
+
+### Contract
+
+The HTLC is constructed by locking an output with the following script:
+
+```asm
+OP_IF
+    OP_SIZE 32 OP_EQUALVERIFY
+    OP_SHA256 <secret_hash> OP_EQUALVERIFY
+    OP_DUP OP_HASH160 <redeem_identity>
+OP_ELSE
+    <timestamp> OP_CHECKLOCKTIMEVERIFY OP_DROP
+    OP_DUP OP_HASH160 <refund_identity>
+OP_ENDIF
+OP_EQUALVERIFY
+OP_CHECKSIG
+```
+
+As required by RFC003, this HTLC uses absolute time locks to check whether `expiry` has been reached.
+Specifically, `OP_CHECKLOCKTIMEVERIFY` (see [BIP65](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki)) is used to compare the time in the block's header to the `expiry` in the contract.
+
+To compute the exact bytes of the contract implementaions should use the following offset table:
+
+| Data              | Position of first byte | Position of last byte | Length | Description                                                                |
+|:------------------|:-----------------------|:----------------------|:-------|:---------------------------------------------------------------------------|
+| `6382012088a820`  | 0                      | 6                     | 7      | `OP_IF OP_SIZE 32 OP_EQUALVERIFY OP_SHA256` + `PUSH32` for the secret_hash |
+| `secret_hash`     | 7                      | 39                    | 32     | See [Parameters](#parameters)                                              |
+| `8876a9`          | 40                     | 42                    | 3      | `OP_EQUALVERIFY OP_DUP OP_HASH160`                                         |
+| `redeem_identity` | 43                     | 74                    | 32     | See [Parameters](#parameters)                                              |
+| `67`              | 75                     | 75                    | 1      | `OP_ELSE`                                                                  |
+| `timestamp`       | 76                     | 79                    | 4      | See [Parameters](#parameters)                                              |
+| `b17576a9`        | 80                     | 83                    | 4      | `OP_CHECKLOCKTIMEVERIFY OP_DROP OP_DUP OP_HASH160`                         |
+| `refund_identity` | 84                     | 115                   | 32     | See [Parameters](#parameters)                                              |
+| `6888ac`          | 116                    | 118                   | 3      | `OP_ENDIF OP_EQUALVERIFY OP_CHECKSIG`                                      |
+
+
+## Execution Phase
+
+### Deployment
+
+The party who is required to deploy the Bitcoin HTLC (the refunder) compiles the contract as described in the previous section.
```

Comment:
I understand why you're not using Alice and Bob, because it would be confusing in the larger context. But I would use "funder"/"deployer" instead of "refunder", since it's the action that will happen most often and puts emphasis on the successful outcome of RFC003.

```suggestion
The party who is required to deploy the Bitcoin HTLC (the funder) compiles the contract as described in the previous section.
```